### ISSUE TITLE
never ending loop with blank file input (patch included)

### ISSUE THREAD

**Issue by asgh** [jquery-form/form#113]
If I have a file input element, but there is no file in it, the plugin says:

[jquery.form] state = uninitialized

In a never ending loop.

The fix is simple:

Change:

```
var fileInputs = $('input:file', this).length > 0;
```

To:

```
var fileInputs = $('input:file[value]', this).length > 0;
```

**Comment by asgh** [comment]
BTW, this may not fix the underlying issue of the never ending loop.

**Comment by asgh** [comment]
Added pull request https://github.com/malsup/form/pull/114

**Comment by winhamwr** [comment]
I was having the same problem and the patch in the pull request fixed my issue.

**Comment by kcivey** [comment]
I'm having the same problem, but even with the patch I'm still getting the infinite loop of `[jquery.form] state = uninitialized`.

**Comment by kcivey** [comment]
Part of my problem was that my response was JSON, which doesn't work well in some browsers when there's a form upload (which has to be handled by a form submission, not ajax). I had to change the content type to text/plain and add dataType 'json' to the ajaxForm options, as described here: http://crazytechthoughts.blogspot.com/2011/01/get-json-data-response-with-jquery-form.html

**Comment by malsup** [comment]
Added proposed fix to v2.91.

**Comment by RobeeeJay** [comment]
Sorry for re-opening this issue, as technically it's a new issue related to the same line of code (which fixing this issue has created)... but...

In Firefox 13, Line 175 of v3.09 of jQuery Form Plugin is returning an empty result which for results in submitting without the file data, or actually even as a file, it just gets sent as an object. :/

This _may_ be a change in Firefox, which is protecting the value of the file input form field from being read by javascript, causing jquery to always think the file field is empty.
